civihr_employee_portal/js/reports.js

```diff
@@ -926,6 +926,74 @@
             }.bind(this)
         });
 
+        $.ajax({
+            type: 'GET',
+            url: this.getJsonUrl() + ( typeof path !== 'undefined' ? path : '' ),
+            success: function(data) {
+                var mappedResults = normaliseSummaryData(data.results);
+
+                replaceSummaryTable(mappedResults, this.$DOM);
+            }.bind(this)
+        });
+
+        function normaliseSummaryData(summary) {
+            var normalisedSummary = {},
+                i;
+
+            // The data can have two formats, depending on particular report:
+            // data = { results: [{ departament: "xxx", count: 1 }, ...]) }
+            // or
+            // data = { results: [{ gender: "aaa" }, { gender: "aaa" }, ...] }
+            //
+            // The code inside if takes care of the first possibility
+            // The else takes care of the second
+            if (summary[0] !== undefined && summary[0].data.count !== undefined) {
+                for(i in summary) {
+                    normalisedSummary[summary[i].data.department] = +summary[i].data.count;
+                }
+            } else {
+                normalisedSummary = summary.map(function (row) {
+                    return { y: row.data.gender, x: 1 }
+                }).reduce(function (accumulator, row) {
+                    if (accumulator[row.y] === undefined) {
+                        accumulator[row.y] = 0;
+                    }
+
+                    accumulator[row.y] += row.x;
+
+                    return accumulator;
+                }, {});
+            }
+
+            return normalisedSummary;
+        }
+
+        function replaceSummaryTable(summary, $DOM) {
+            var $countRow = $('<tr />'),
+                $headerRow = $('<tr />'),
+                $percentageRow = $('<tr />'),
+                summaryKeys = Object.keys(summary),
+                total = summaryKeys.reduce(function(accumulator, key) { return accumulator + summary[key]; }, 0);
```

Comment:
Agree. To be honest I still haven't decided about multi-vars declaration style

This:
```js
var a = 'a';
var b = 1;
var c = {};
```
or this
```js
var a = 'a', b = 1, c = {};
```
or this
```js
var a = 'a',
    b = 1,
    c = {};
```
or even this (mimics hoisting)
```js
var a, b, c;

a = 'a';
b = 1;
c = {};
```
I usually tend to use the first format, but i'm a bit all over the place with this